### ISSUE TITLE
Enum values as column defaultValue doesn`t work anymore

### ISSUE THREAD

**Issue by oyvindvol** [dotnet/efcore#9685]
I am having trouble with running migrations that are using enums as default values after upgrading to EntityFrameworkCore 2.0.0. The problem seems to be that  EF is no longer trying to use the int-value of the enum. Example migration:

```
migrationBuilder.AddColumn<int>(
                name: "RosColor",
                table: "Ross",
                nullable: false,
                defaultValue: RosColor.Red);
```
Enum:
```
public enum RosColor
    {
        Red = 0,
        Yellow = 1,
        Green = 2
    }
```
Error:
``ERROR!The name "Red" is not permitted in this context. Valid expressions are constants, constant expressions, and (in some contexts) variables. Column names are not permitted.``

In EF Core 1.1.2 this error did not occur, and the default value would be `0`. Does anyone know how to deal with this?

### Steps to reproduce
Create a column with default value as an enum with an associated int value. Run migrations. See that an error occurs when trying to set the default value to the enum rather than the enums integer value.

### Expected results
The column default value is set to the integer value like in EF core 1.1.2.

### Further technical details
EF Core version: 2.0.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Win10
IDE: Visual Studio 2017

**Comment by ajcvickers** [comment]
Hi @oyvindvol. We are gathering information on the use of EF Core pre-release builds. You reported this issue shortly after the release of 2.0.0 RTM. It would be really helpful if you could let us know:

- Did you consider testing your code against the pre-release builds?
- Was there anything blocking you from using pre-release builds?
- What do you think could make it easier for you to use pre-release builds in the future?

Thanks in advance for any feedback. Hopefully this will help us to increase the value of pre-release builds going forward.

**Comment by bricelam** [comment]
I am not able to reproduce this using version 2.0.0. @oyvindvol, could you share your full code so we can investigate further?

**Comment by oyvindvol** [comment]
@bricelam The migration with the enum default value was created in an earlier version of EF. New migrations are getting default value equal to the int value of the enum. However, the problem was that the migrations would not be run with 2.0.0.

I solved the issue by editing the migration, casting the enum default value to int.

**Comment by ajcvickers** [comment]
Since this was fixed in #4061, which shipped in 1.0.0 RC2, it means that migrations that have this came from a pre-release version of EF. The guidance here is to manually update the migrations to use the underlying type.
